data-prepper-plugins/s3-source/src/main/java/org/opensearch/dataprepper/plugins/source/s3/SqsWorker.java

```diff
@@ -228,6 +231,10 @@ && isEventBridgeEventTypeCreated(parsedMessage)) {
         LOG.info("Received {} messages from SQS. Processing {} messages.", s3EventNotificationRecords.size(), parsedMessagesToRead.size());
         
         for (ParsedMessage parsedMessage : parsedMessagesToRead) {
+            sqsMessageDelayTimer.record(Duration.between(
```

Comment:
This change is untested. We can use a Mockito `inOrder` to verify that this happens before or after a certain part in the tests. I'm guessing there is an existing test that could be updated to use inOrder to ensure this behavior remains consistent.

e.g.

```
InOrder inOrder = inOrder(s3Service, sqsMessageDelayTimer);
inOrder.verify(sqsMessageDelayTimer).record(...);
inOrder.verify(s3Service).addS3Object(...);
```

Reply:
Makes sense I can make that change